src/io.jl

```diff
@@ -99,8 +99,8 @@ end
 decode_null(::Decoder) = nothing
 decode_boolean(decoder::Decoder) = read(decoder.stream, Bool)
 
-decode_int(decoder::Decoder) = _decode_zigzag(_decode_varint(decoder.stream, Int32))
-decode_long(decoder::Decoder) = _decode_zigzag(_decode_varint(decoder.stream, Int64))
+decode_int(decoder::Decoder)::Int32 = _decode_zigzag(_decode_varint(decoder.stream, Int32))
+decode_long(decoder::Decoder)::Int64 = _decode_zigzag(_decode_varint(decoder.stream, Int64))
```

Comment:
Did you add these type assertions for the purpose of documentation?

Reply:
If you actually need the type coersion that this would imply, I would rather tweak the definition of `_decode_varint`, which I think might be the result of the type instability. (Replace the last line with `convert(T, n)`.)

Reply:
Looking closer at the definition of `_decode_varint`, I can't see how either it or `_decode_zigzag` can fail to return the desired type. Still, I think it would be regarded as suboptimal form to rely on a type coersion on the return value in this way (vs. putting the conversion in the body).